cmake/kokkos_backends.cmake

```
@@ -10,9 +10,15 @@ MACRO(CHECK_KOKKOS_BACKEND BE)
 ENDMACRO(CHECK_KOKKOS_BACKEND)
 
 CHECK_KOKKOS_BACKEND(SERIAL)
-CHECK_KOKKOS_BACKEND(PTHREAD)
+CHECK_KOKKOS_BACKEND(THREADS)
 CHECK_KOKKOS_BACKEND(OPENMP)
 CHECK_KOKKOS_BACKEND(OPENMPTARGET)
 CHECK_KOKKOS_BACKEND(CUDA)
 CHECK_KOKKOS_BACKEND(HIP)
 CHECK_KOKKOS_BACKEND(SYCL)
+
+# for backward compatibility.  can be dropped when requiring Kokkos 3.6
+IF (Kokkos_ENABLE_PTHREAD)
```

`Kokkos_ENABLE_PTHREAD` is not defined if deprecated code is disabled
```suggestion
IF (Kokkos_ENABLE_PTHREAD || Kokkos_ENABLE_THREADS)
```

---

In that case should we not be fine with `CHECK_KOKKOS_BACKEND(THREADS)`?

---

Looks like it is fine with this `IF (Kokkos_ENABLE_PTHREAD)` on line 21 + `CHECK_KOKKOS_BACKEND(THREADS)` above.

---

Oh right